luda-editor/luda-editor-client/src/app/sequence_list/list_item.rs

```diff
@@ -0,0 +1,84 @@
+use super::{
+    events::SequenceListEvent,
+    rounded_rectangle::RoundedRectangleColor,
+    types::{RenderingTreeRow, RenderingTreeRows, SequenceLoadStateDetail},
+    SequenceList, BUTTON_HEIGHT, MARGIN,
+};
+use crate::app::sequence_list::SPACING;
+use namui::{render, Wh};
+
+impl SequenceList {
+    pub fn render_list_item(&self, width: f32, title: &String) -> RenderingTreeRow {
+        let path = format!("sequence/{}", title);
+        let element_width = width - 2.0 * MARGIN;
+        let button_wh = Wh {
+            width: element_width,
+            height: BUTTON_HEIGHT,
+        };
+        let title_load_state = self.sequence_load_state_map.get(&path);
+        let is_open = title_load_state.is_some();
+        let mut elements: Vec<RenderingTreeRow> = Vec::new();
+
+        elements.push(RenderingTreeRow::new(
+            render![
+                self.render_rounded_rectangle(button_wh, RoundedRectangleColor::DarkGray)
+                    .with_mouse_cursor(namui::MouseCursor::Pointer)
+                    .attach_event(move |builder| {
+                        let path = path.clone();
+                        builder.on_mouse_down(Box::new(move |_| match is_open {
+                            true => {
+                                let path = path.clone();
+                                namui::event::send(
+                                    SequenceListEvent::SequenceLoadStateUpdateEvent {
+                                        path,
+                                        state: None,
+                                    },
+                                );
+                            }
+                            false => {
+                                let path = path.clone();
+                                namui::event::send(SequenceListEvent::SequenceLoadEvent { path });
+                            }
+                        }))
```

Comment:
Is there a good way to remove `let path = path.clone();`?

Reply:
You cannot remove 'clone' for move String to closure because String didn't implement Copy trait.  [Reference](http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/std/marker/trait.Copy.html#when-cant-my-type-be-copy)

you used `let path = path.clone()` 3 times, but it could be 2 times.

```rust
                        let path = path.clone();
                        builder.on_mouse_down(Box::new(move |_| {
                            let path = path.clone();
                            match is_open {
                                true => {
                                    namui::event::send(
                                        SequenceListEvent::SequenceLoadStateUpdateEvent {
                                            path,
                                            state: None,
                                        },
                                    );
                                }
                                false => {
                                    namui::event::send(SequenceListEvent::SequenceLoadEvent { path });
                                }
                            }))
```